### DISCUSSION TITLE
Allow inverting the state of a cover entity

### DISCUSSION THREAD

**Discussion by emontnemery** [home-assistant/architecture#983] [Entity Models]
## Proposal

Add an entity option to invert the state of a cover; an `open` cover will be `closed` and a `closed` cover will be `open`. Also, the cover's position should be inverted.

### Complications
Covers have two properties corresponding to their open/closed state:
- `is_closed` - Returns a bool
- `current_cover_position` - Returns an integer 0..100 where where 0 means closed and 1..100 means open

Just inverting `is_closed` won't work because position 0 will then mean open and 1..100 will mean closed.

Possible way forward could be to:
- Let the base class calculate `is_closed` based on the `current_cover_position`
- Deprecate `is_closed`, covers with a binary state would instead implement `current_cover_position` to return either 0 or 100

**Comment by thecode** [comment]
I am just noting here that in Shelly Gen2 (Gen1 is based on closed=0) we report the state from the device, not based on position:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/shelly/cover.py#L163
This is aligned with the device docs https://shelly-api-docs.shelly.cloud/gen2/ComponentsAndServices/Cover#status because the device can still know it is closed or open while it can't report its position.

While this is not common and may happen due to a problem, I have already seen that, the device lost position, reported that the position is unknown but the cover is closed (since the last direction moved until consumption was 0), you can claim that the device also knows that the position is 0, I don't know the reasons to report closed without a percentage but they may be logic behind that.

I think that a default `is_closed` method can calculate the position based on percentage, but an integration should be allowed to override it.
I understand that this conflicts with a future idea to invert the cover, but this discussion is about the `is_closed` and if a future idea is needed it would be better to create a discussion for it and discuss how to handle it.

**Comment by rrooggiieerr** [comment]
I think this would be a neat addition to the cover entity. My use case would be a projector screen, which is closed when retracted while open when pulled out. Totally opposite the behaviour of the cover entity.

**Comment by TimonTrukker** [comment]
Slightly related, I have an awning which has correct states (closed means not deployed, open means deployed) and correct positions (0 means not deployed and 100 means fully deployed), but the slider to set a specific condition shows what I interpret as a deployed cover when the position is 0 and what I interpret as a closed cover when the position is 100. So a method to invert that behavior would be awesome.

**Comment by heindrichpaul** [comment]
@rrooggiieerr I have the same problem where the technical state is different from what feels different. I also logged a feature request [here](https://github.com/orgs/home-assistant/discussions/1844#discussioncomment-15146695) and I am working on a PR that would implement this. I have also posted on another cover architectural discussion about this.

**Reply by BradleyFord** [reply]
yes, please I have the same issue 

Units that have the motor on the left show the reverse status, units with motor on the right show correctly... :(

**Comment by warksit** [comment]
## Add `projector_screen` device class for cover domain

### Problem

Projector screens have inverted physical behavior compared to typical covers:
- **"Open"** = deploy the screen (moves **down**, in use)
- **"Close"** = retract the screen (moves **up**, not in use)

Currently, users must choose between:
- Using `blind`/`shade` device class → wrong directional arrows (↑ for open, ↓ for close)
- Reporting inverted state → confuses "open"/"closed" semantics

### Proposed Solution

Add `projector_screen` as a new cover device class with:
- Swapped directional icons for open/close buttons
- Appropriate entity state icons

### Affected Devices

This issue affects various motorized projector screen brands and DIY setups:
- **XY Screens** (RS485 interface)
- **Celexon** (Battery/RF controlled)
- **RFXCom AC114** projector screens
- **DIY setups** using Shelly, Sonoff/Tasmota, Broadlink, and ESPHome controllers

Community threads reporting this issue:
- [Open/close direction for projector screens](https://community.home-assistant.io/t/open-close-direction-for-projector-screens/464292)
- [Motor projector screen as cover component](https://community.home-assistant.io/t/motor-projector-screen-as-cover-component/84583)
- [Projector Screen with Shelly](https://community.home-assistant.io/t/projector-screen-with-shelly/352355)
- [Inverting rfxtrx cover (RFXCom AC114 projector screen)](https://community.home-assistant.io/t/inverting-rfxtrx-cover-rfxcom-ac114-projector-screen/84081)

### Implementation (Tested Locally)

#### Core Changes

**`homeassistant/components/cover/__init__.py`**
```python
class CoverDeviceClass(StrEnum):
    # ... existing entries ...
    PROJECTOR_SCREEN = "projector_screen"
```

**`homeassistant/components/cover/icons.json`**
```json
"projector_screen": {
  "default": "mdi:projector-screen",
  "state": {
    "closed": "mdi:projector-screen-off",
    "closing": "mdi:arrow-up-box",
    "opening": "mdi:arrow-down-box"
  }
}
```

#### Frontend Changes

**`src/common/entity/cover_icon.ts`**
```typescript
export const computeOpenIcon = (stateObj: HassEntity): string => {
  switch (stateObj.attributes.device_class) {
    case "projector_screen":
      return mdiArrowDown; // "Open" means deploy (go down)
    // ... existing cases ...
  }
};

export const computeCloseIcon = (stateObj: HassEntity): string => {
  switch (stateObj.attributes.device_class) {
    case "projector_screen":
      return mdiArrowUp; // "Close" means retract (go up)
    // ... existing cases ...
  }
};
```

### Result

| State | Entity Icon | Open Button | Close Button |
|-------|-------------|-------------|--------------|
| Closed (retracted) | `mdi:projector-screen-off` | ↓ (active) | ↑ (disabled) |
| Opening (deploying) | `mdi:arrow-down-box` | ↓ (disabled) | ↑ (active) |
| Open (deployed) | `mdi:projector-screen` | ↓ (disabled) | ↑ (active) |
| Closing (retracting) | `mdi:arrow-up-box` | ↓ (active) | ↑ (disabled) |

### Existing Patterns

Similar device classes exist for covers with non-standard physical orientations:
- `awning`, `curtain`, `door`, `gate` → horizontal expand/collapse icons

**Comment by luka6000** [comment]
this is a very much needed change for covers.
Take blinds: fully extended blinds are physically "open" but the window is covered by blinds. So when I say "close blinds" I actually mean "cover window with blinds" which physically fully opens/extends/deploys the blinds. "Open blinds" command means to physically close/stow away the blinds because I actually want to uncover the window. Currently it's possible with cover template and you end up with two entities with opposite state for the same device.

**Comment by warksit** [comment]
I have a minimal implementation ready for `projector_screen` as a new cover device class, addressing the inverted direction issue for this specific device type.

**What the implementation does:**
- Adds `PROJECTOR_SCREEN` to `CoverDeviceClass` enum
- Adds state icons with inverted directional arrows (`mdi:arrow-down-box` for opening/deploying, `mdi:arrow-up-box` for closing/retracting)
- Frontend: swaps open/close button arrows to match physical movement
- No behavioral changes to the cover domain itself

**PRs (closed pending this discussion's approval):**
- Core: https://github.com/home-assistant/core/pull/163618
- Frontend: https://github.com/home-assistant/frontend/pull/29738

This has been an open need for over two years now, with community interest across multiple integrations (XY Screens, Celexon, RFXCom, DIY setups). Is there anything blocking approval or anything needed to move this forward?

**Reply by MartinHjelmare** [reply]
First of all, you should open a separate architecture discussion for the suggestion to add a new device class. It's not the same as the suggestion in this architecture discussion.

In the new discussion, please list all the current core integrations that can leverage the new device class. Don't include integrations that can support anything like Template, MQTT, esphome, helpers etc.

**Reply by warksit** [reply]
Thanks @MartinHjelmare. I've opened a separate discussion for the `projector_screen` device class proposal: https://github.com/home-assistant/architecture/discussions/1342
